pallets/scoring/src/lib.rs

```diff
@@ -0,0 +1,337 @@
+		/// Invalid Identifer Length
+		InvalidIdentifierLength,
+		// Invalid digest
+		InvalidDigest,
+		// Invalid creator signature
+		InvalidSignature,
+		// Invalid Rating Identifier
+		InvalidRatingIdentifier,
+		// Transaction already rated
+		TransactionAlreadyRated,
+		// Invalid rating value - should be between 1 and 50
+		InvalidRatingValue,
+		// Exceeds the maximum allowed entries in a single transaction
+		TooManyJournalEntries,
+		// Invalid entity signature
+		InvalidEntitySignature,
+		//Stream digest is not unique
+		DigestAlreadyAnchored,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		///
+		/// Create a new rating identifier and associates it with its
+		/// controller. The controller (issuer) is the owner of the identifier.
+		///
+		/// * origin: the identity of the Transaction Author. Transaction author
+		///   pays the transaction fees
+		/// * tx_journal: the incoming rating entry.
+		/// * `authorization`: The authorization ID of the delegate who is
+		///   allowed to perform this action.
+		#[pallet::call_index(0)]
+		#[pallet::weight({0})]
+		pub fn entries(
+			origin: OriginFor<T>,
+			journal: RatingInputOf<T>,
+			authorization: AuthorizationIdOf,
+		) -> DispatchResult {
+			let author = <T as Config>::EnsureOrigin::ensure_origin(origin)?.subject();
+
+			let registry_id =
+				pallet_registry::Pallet::<T>::is_a_delegate(&authorization, author.clone(), None)
+					.map_err(<pallet_registry::Error<T>>::from)?;
+
+			ensure!(
+				(journal.entry.rating >= T::MinScoreValue::get() &&
+					journal.entry.rating <= T::MaxScoreValue::get()),
+				Error::<T>::InvalidRatingValue
+			);
+			ensure!(
+				!<JournalHashes<T>>::contains_key(&journal.digest),
+				Error::<T>::DigestAlreadyAnchored
+			);
+
+			let identifier = Ss58Identifier::to_scoring_id(&(&journal.digest).encode()[..])
+				.map_err(|_| Error::<T>::InvalidIdentifierLength)?;
+
+			ensure!(
+				!<TidEntries<T>>::contains_key(&journal.entry.tid, &journal.entry.rating_type),
+				Error::<T>::TransactionAlreadyRated
+			);
+
+			let block_number = frame_system::Pallet::<T>::block_number();
+
+			<Journal<T>>::insert(
+				&identifier,
+				&journal.entry.rating_type,
+				RatingEntryOf::<T> {
+					entry: journal.entry.clone(),
+					digest: journal.digest,
+					created_at: block_number,
+					registry: registry_id,
+					creator: author.clone(),
+				},
+			);
+			<JournalHashes<T>>::insert(&journal.digest, ());
+			<TidEntries<T>>::insert(
+				&journal.entry.tid,
+				&journal.entry.rating_type,
+				&journal.entry.entity,
+			);
+
+			Self::deposit_event(Event::JournalEntry {
+				identifier,
+				entity: journal.entry.entity.clone(),
+				author: author.clone(),
+			});
+
+			Self::aggregate_score(&journal.entry);
+
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn aggregate_score(entry: &RatingDetailsOf<T>) {
+		if let Some(mut aggregate) = <Scores<T>>::get(&entry.entity, &entry.rating_type) {
+			aggregate.count.saturating_inc();
```

Comment:
We need to check the rating type and act accordingly. A debit type should be added, and a credit type should reduce the existing score. For the credit type, we also need to ensure that the current value exceeds the credit transaction value.

Reply:
both count and value